bin/pre-commit-hook.js

```diff
@@ -28,3 +32,28 @@ if ( lintResult.status ) {
 		'repeat the commit command with --no-verify to avoid this check.' );
 	process.exit( 1 );
 }
+
+
+// run prettier for any files in the commit that have @format within their first docblock
+const formatRegex = /@format/;
+const firstDocBlockStart = /\/\*\*/;
+const firstDocBlockEnd = /\*\//;
+
+files
+	.map( file => path.join( __dirname, '../', file ) )
+	.forEach( file => {
+		fs.readFile( file, 'utf8', ( err, data ) => {
+			const firstDocBlockStartIndex = _.get( firstDocBlockStart.exec( data ), 'index' );
+			const firstDocBlockEndIndex = _.get( firstDocBlockEnd.exec( data ), 'index' );
+			const firstDocBlockText =
+				_.isInteger( firstDocBlockStartIndex ) &&
+				_.isInteger( firstDocBlockEndIndex ) &&
+				data.substring( firstDocBlockStartIndex, firstDocBlockEndIndex + 2 );
+
+			if ( firstDocBlockText && formatRegex.test( firstDocBlockText ) ) {
+				const formattedData = prettier.format( data, {} );
+				fs.writeFileSync( file, formattedData );
+				execSync( `git add ${ file }` );
+			}
```

Comment:
ah, i see! that explains the "look for anything except for docblocks" part of the regex.  Thats the part I was having trouble with.  Thanks for diving into this and making it a learning experience for me!

I'd still argue that the regex-only method would be harder to maintain/understand than the verbose-but-simple logic I've currently got that goes something like this:
```js
startIndex = ...
endIndex = ...
if ( /@format/.test( file.substring( startIndex, endIndex+2 ) ) {
  launchThePrettyCannons();
}
```